builtin/credential/cert/path_login.go

```diff
@@ -507,17 +508,29 @@ func (b *backend) matchesCertificateExtensions(clientCert *x509.Certificate, con
 	// including its ASN.1 type tag bytes. For the sake of simplicity, assume string type
 	// and drop the tag bytes. And get the number of bytes from the tag.
 	clientExtMap := make(map[string]string, len(clientCert.Extensions))
+	hexExtMap := make(map[string]string, len(clientCert.Extensions))
+
 	for _, ext := range clientCert.Extensions {
 		var parsedValue string
 		asn1.Unmarshal(ext.Value, &parsedValue)
```

Comment:
Kinda odd we ignore the unmarshalling error here?

Reply:
Yes, but this is existing behavior; it results in `map[<oid>] = ""` in the map, which you _could_ match on in the API by providing just the OID.

Reply:
Could we codify it explicitly so I don't break existing customers, in the future thinking I'm fixing a bug :D

Reply:
Interestingly, we do have:

```
			testAccStepCert(t, "web", ca, "foo", allowed{ext: "2.1.1.1:"}, false),
			testAccStepLoginInvalid(t, connState),
			testAccStepCert(t, "web", ca, "foo", allowed{ext: "2.1.1.1:,2.1.1.2:*"}, false),
			testAccStepLoginInvalid(t, connState),
```

as test cases, which seems to suggest this should be rejected for login, but not from the API, which is weird. I'm not entirely sure why this wouldn't work though, tbh.

Reply:
Ah, but these are string extensions. You'd need to do this:

```
testAccStepCert(t, "web", ca, "foo", allowed{ext: "2.5.29.17:"}, false),
testAccStepLogin(t, connState),
```

which seems to succeed.

Reply:
@rculpepper I'd probably add this as an explicit test case btw. :-)

Reply:
Okay, if we can't change this behavior, or it is what we want, could we explicitly, trap the error and set the value to "" so the code is explicit about the behavior we want.